src/sonic-yang-mgmt/sonic_yang.py

```diff
@@ -0,0 +1,450 @@
+       except Exception as e:
+            print(e)
+            return False
+       self.root = node
+       return True
+
+    """
+    get module name from xpath
+    input:    xpath
+    returns:  module name
+    """
+    def get_module_name(self, xpath):
+        module_name = xpath.split(':')[0].strip('/')
+        return module_name
+
+    """
+    get_module(): get module object from Yang module name
+    input:   yang module name
+    returns: Schema_Node object
+    """
+    def get_module(self, module_name):
+        mod = self.ctx.get_module(module_name)
+        return mod
+
+    """
+    load_data_model(): load both Yang module file and data json file
+    input:   yang directory, yang file and data file
+    returns: (context, root)
+    """
+    def load_data_model (self, yang_dir, yang_file, data_file, output):
+        if (self.ctx is None):
+            self.ctx = ly.Context(yang_dir)
+
+        if self.module is  None:
+            module = self.ctx.load_module(yang_file)
+            self.module = module
+
+        try:
+            node = self.ctx.parse_data_path(data_file, ly.LYD_JSON, ly.LYD_OPT_CONFIG | ly.LYD_OPT_STRICT)
+        except Exception as e:
+            node = None
+            print(e)
+
+        self.root = node
+        return (self.ctx, node)
+
+    """
+    print_data_mem():  print the data tree
+    input:  option:  "JSON" or "XML"
+    """
+    def print_data_mem (self, option):
+        if (option == "JSON"):
+            mem = self.root.print_mem(ly.LYD_JSON, ly.LYP_WITHSIBLINGS | ly.LYP_FORMAT)
+        else:
+            mem = self.root.print_mem(ly.LYD_XML, ly.LYP_WITHSIBLINGS | ly.LYP_FORMAT)
+
+        print("======================= print data =================")
+        print(mem)
+
+    """
+    save_data_file_json(): save the data tree in memory into json file
+    input: outfile - full path of the file to save the data tree to
+    """
+    def save_data_file_json(self, outfile):
+        mem = self.root.print_mem(ly.LYD_JSON, ly.LYP_FORMAT)
+        with open(outfile, 'w') as out:
+            json.dump(mem, out, indent=4)
+
+    """
+    get_module_tree(): get yang module tree in JSON or XMAL format
+    input:   module name
+    returns: JSON or XML format of the input yang module schema tree
+    """
+    def get_module_tree(self, module_name, format):
+        result = None
+        module = self.ctx.get_module(str(module_name))
+        if (module is not None):
+            if (format == "XML"):
+                #libyang error with format
+                result = module.print_mem(ly.LYD_JSON, ly.LYP_FORMAT)
+            else:
+                result = module.print_mem(ly.LYD_XML, ly.LYP_FORMAT)
+
+        return result
+
+    """
+    validate_data(): validate data tree
+    input:
+           node:   root of the data tree
+           ctx:    context
+    returns:  True - success   False - failed
+    """
+    def validate_data (self, node, ctx):
```

Comment:
if node = self.node and ctx = self.ctx (always),  then I think we should not pass it.

Reply:
Added default auguments